packages/mdl-base/README.md

```diff
@@ -0,0 +1,241 @@
+      BUTTON: 'my-component__button',
+      TOGGLED: 'my-component--toggled'
+    };
+  }
+
+  static get defaultAdapter() {
+    return {
+      toggleClass: (/* className: string */) => {},
+      registerBtnClickHandler: (/* handler: Function */) => {},
+      deregisterBtnClickHandler: (/* handler: Function */) => {}
+    };
+  }
+
+  constructor(adapter) {
+    super(Object.assign(MyFoundation.defaultAdapter, adapter));
+    const {TOGGLED} = MyFoundation.cssClasses;
+    this.clickHandler_ = () => this.adapter_.toggleClass(TOGGLED);
+  }
+
+  init() {
+    this.adapter_.registerBtnClickHandler(this.clickHandler_);
+  }
+
+  destroy() {
+    this.adapter_.deregisterBtnClickHandler(this.clickHandler_);
+  }
+}
+```
+
+#### Static Getters
+
+The static getters specify constants that can be used within the foundation class, its component, and by 3rd-party code. _It's important to remember to always put constants into these getters_. This will ensure your component can interop in as many environments as possible, including those where CSS classes need to be overwritten by the host library (e.g., Closure Stylesheets), or strings need to be modified (for i18n, for example).
+
+Note that you do not have to explicitly provide getters for constants if your component has none.
+
+The getters which should be provided are specified below:
+
+| getter | description |
+| --- | --- |
+| cssClasses | returns an object where each key identifies a css class that some code will rely on. |
+| strings | returns an object where each key identifies a string constant, e.g. `ARIA_ROLE` |
+| numbers | returns an object where each key identifies a numeric constant, e.g. `TRANSITION_DELAY_MS` |
+| defaultAdapter | returns an object specifying the shape of the adapter. Can be used as sensible defaults for an adapter as well as a way to specify your adapter's "schema" |
+
+#### Lifecycle Methods
+
+Each foundation class has two lifecycle methods: `init()` and `destroy()`, which are described below:
+
+| method | time of invocation | use case |
+| --- | --- | --- |
+| init() | called by a host class when a component is ready to be initialized | add event listeners, query for info via adapters, etc. |
+| destroy() | called by a host class when a component is no longer in use | remove event listeners, reset any transient state, etc. |
+
+> Please note: _the lifecycle methods are **not** a safe place to perform DOM reads/writes that would invalidate layout or cause a repaint_. If this needs to be done within these methods, it should be put into a `requestAnimationFrame()` call so that it's synchronized with the browser's refresh cycle and does not [cause jank](http://www.html5rocks.com/en/tutorials/speed/rendering/).
+
+### MDLComponent
+
+MDLComponent provides the basic mechanisms for implementing component classes.
+
+```javascript
+import MyComponentFoundation from './foundation';
+
+export default class MyComponent extends MDLComponent {
```

Comment:
Using `MyComponent` in this example and below might be confusing because MyComponent is the name of the component (though the term component wouldn't really appear in a component name). Or at least it seemed a bit weird to see the words ComponentFoundation next to each other so often. So I wonder if `MyButton` or equivalent would work better?
If it seems fine, just ignore this comment.